process/nettisa.hpp

```diff
@@ -0,0 +1,201 @@
+
+   float mean;
+   float min;
+   float max;
+   float stdev;
+   float kurtosis;
+   float root_mean_square;
+   float average_dispersion;
+   float mean_scaled_time;
+   float mean_difftimes;
+   float min_difftimes;
+   float max_difftimes;
+   float time_distribution;
+   float switching_ratio;
+
+   uint16_t prev_payload;
+   long prev_time;
+    
+   RecordExtNETTISA() : RecordExt(REGISTERED_ID)
+   {
+       mean = 0;
+       min = 0;
+       max = 0;
+       stdev = 0;
+       kurtosis = 0;
+       root_mean_square = 0;
+       average_dispersion = 0;
+       mean_scaled_time = 0;
+       mean_difftimes = 0;
+       min_difftimes = 0;
+       max_difftimes = 0;
+       time_distribution = 0;
+       switching_ratio = 0;
+    
+       prev_payload = 0;
+       prev_time = 0;
+   }
+
+#ifdef WITH_NEMEA
+   virtual void fill_unirec(ur_template_t *tmplt, void *record)
+   {
+      ur_set(tmplt, record, F_NTS_MEAN, mean);
+      ur_set(tmplt, record, F_NTS_MIN, min);
+      ur_set(tmplt, record, F_NTS_MAX, max);
+      ur_set(tmplt, record, F_NTS_STDEV, stdev);
+      ur_set(tmplt, record, F_NTS_KURTOSIS, kurtosis);
+      ur_set(tmplt, record, F_NTS_ROOT_MEAN_SQUARE, root_mean_square);
+      ur_set(tmplt, record, F_NTS_AVERAGE_DISPERSION, average_dispersion);
+      ur_set(tmplt, record, F_NTS_MEAN_SCALED_TIME, mean_scaled_time);
+      ur_set(tmplt, record, F_NTS_MEAN_DIFFTIMES, mean_difftimes);
+      ur_set(tmplt, record, F_NTS_MIN_DIFFTIMES, min_difftimes);
+      ur_set(tmplt, record, F_NTS_MAX_DIFFTIMES, max_difftimes);
+      ur_set(tmplt, record, F_NTS_TIME_DISTRIBUTION, time_distribution);
+      ur_set(tmplt, record, F_NTS_SWITCHING_RATIO, switching_ratio);
+   }
+
+   const char *get_unirec_tmplt() const
+   {
+      return NETTISA_UNIREC_TEMPLATE;
+   }
+#endif // ifdef WITH_NEMEA
+
+   virtual int fill_ipfix(uint8_t *buffer, int size)
+   {
+      uint16_t len_mean               = sizeof(mean);
+      uint16_t len_min                = sizeof(min);
+      uint16_t len_max                = sizeof(max);
+      uint16_t len_stdev              = sizeof(stdev);
+      uint16_t len_kurtosis           = sizeof(kurtosis);
+      uint16_t len_root_mean_square   = sizeof(root_mean_square);
+      uint16_t len_average_dispersion = sizeof(average_dispersion);
+      uint16_t len_mean_scaled_time   = sizeof(mean_scaled_time);
+      uint16_t len_mean_difftimes     = sizeof(mean_difftimes);
+      uint16_t len_min_difftimes      = sizeof(min_difftimes);
+      uint16_t len_max_difftimes      = sizeof(max_difftimes);
+      uint16_t len_time_distribution  = sizeof(time_distribution);
+      uint16_t len_switching_ratio   = sizeof(switching_ratio);
+      int pos = 0;
+
+      if (len_mean + len_min + len_max + len_stdev + len_kurtosis + len_root_mean_square + len_average_dispersion + len_mean_scaled_time + len_mean_difftimes + len_min_difftimes + len_max_difftimes + len_time_distribution + len_switching_ratio > size) {
```

Comment:
Prepsal bych to na neco hezciho, treba:

```suggestion
  constexpr size_t get_ipfix_size() const noexcept
  {
      return sizeof(mean) + sizeof(min) + ...;
  }
  
  virtual int fill_ipfix(uint8_t *buffer, int available_ipfix_size)
   {
       size_t required_ipfix_size = get_ipfix_size();
       if (required_ipfix_size > available_ipfix_size) {
           return -1;
       }
   }
```